src/qt/walletview.cpp

```diff
@@ -205,23 +207,37 @@ void WalletView::gotoVerifyMessageTab(QString addr)
         signVerifyMessageDialog->setAddress_VM(addr);
 }
 
-void WalletView::gotoLoadPSBT()
+void WalletView::gotoLoadPSBT(bool from_clipboard)
 {
-    QString filename = GUIUtil::getOpenFileName(this,
-        tr("Load Transaction Data"), QString(),
-        tr("Partially Signed Transaction (*.psbt)"), nullptr);
-    if (filename.isEmpty()) return;
-    if (GetFileSize(filename.toLocal8Bit().data(), MAX_FILE_SIZE_PSBT) == MAX_FILE_SIZE_PSBT) {
-        Q_EMIT message(tr("Error"), tr("PSBT file must be smaller than 100 MiB"), CClientUIInterface::MSG_ERROR);
-        return;
+    std::string data;
+
+    if (from_clipboard) {
+        std::string raw = QApplication::clipboard()->text().toStdString();
+        bool invalid;
+        data = DecodeBase64(raw, &invalid);
+        if (invalid) {
+            Q_EMIT message(tr("Error"), tr("Unable to decode PSBT from clipboard (invalid base64)"), CClientUIInterface::MSG_ERROR);
+            return;
+        }
+    } else {
+        QString filename = GUIUtil::getOpenFileName(this,
+            tr("Load Transaction Data"), QString(),
+            tr("Partially Signed Transaction (*.psbt)"), nullptr);
+        if (filename.isEmpty()) return;
+        if (GetFileSize(filename.toLocal8Bit().data(), MAX_FILE_SIZE_PSBT) == MAX_FILE_SIZE_PSBT) {
+            Q_EMIT message(tr("Error"), tr("PSBT file must be smaller than 100 MiB"), CClientUIInterface::MSG_ERROR);
+            return;
+        }
+        std::ifstream in(filename.toLocal8Bit().data(), std::ios::binary);
+        std::stringstream sstr;
+        sstr << in.rdbuf();
```

Comment:
nit: this refactor in how the file is read might deserve its own commit.

Reply:
I'm not sure how much sense that change makes in isolation -- it was forced by the clipboard change, in which it is included, because when I lifted the definition of 'data' to an outer scope, I couldn't figure out how to avoid changing the approach at the same time (since I was no longer able to use a non-default constructor of std::string.)

Reply:
... And actually the refactor isn't necessary, I can do `data = std::string(std::istreambuf_iterator<char>{in}, {});` and move-assignment should cause the copy to be elided so it's free. So I'll just change it to that.